src/battle_script_commands.c

```diff
@@ -3647,6 +3647,22 @@ void SetMoveEffect(bool32 primary, u32 certain)
                 BattleScriptPush(gBattlescriptCurrInstr + 1);
                 gBattlescriptCurrInstr = BattleScript_DoubleShockRemoveType;
                 break;
+            case MOVE_EFFECT_DIRE_CLAW:
+                if (gBattleMons[gEffectBattler].status1)
+                {
+                    gBattlescriptCurrInstr++;
+                }
+                else
+                {
+                DIRE_CLAW_STATUS_PICK:
+                    gBattleScripting.moveEffect = Random() % 6;
+                    if (gBattleScripting.moveEffect == 0
+                     || gBattleScripting.moveEffect == MOVE_EFFECT_BURN
+                     || gBattleScripting.moveEffect == MOVE_EFFECT_FREEZE)
+                        goto DIRE_CLAW_STATUS_PICK;
```

Comment:
I'd potentially write this more like:
```c
static const u8 sDireClawStatuses[] = { MOVE_EFFECT_POISON, MOVE_EFFECT_PARALYSIS, MOVE_EFFECT_SLEEP };

gBattleScripting.moveEffect = sDireClawStatuses[Random() % 3];
```

Reply:
Ohh! :eyes:

I did want to use an array at first, but that wasn't working too well because back then I was trying to apply the `STATUS1` directly.

I'll test that out, thanks!

Reply:
Tested it out after fighting with Shell Trap this entire week. Sadly, it doesn't work 😔